src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -488,8 +490,14 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
                 return null;
             }
 
-            return TupleTypeSymbol.Create(locationOpt: null, elementTypes: typesBuilder.ToImmutableAndFree(), elementLocations: default(ImmutableArray<Location>),
-                                    elementNames: default(ImmutableArray<string>), compilation: compilation, diagnostics: diagnostics);
+            return TupleTypeSymbol.Create(
+                locationOpt: null,
+                elementTypes: typesBuilder.ToImmutableAndFree(),
+                elementLocations: default(ImmutableArray<Location>),
+                elementNames: default(ImmutableArray<string>),
+                compilation: compilation,
+                diagnostics: diagnostics,
+                shouldCheckConstraints: false);
```

Comment:
I think the deconstruction tests in this PR cover this. 
`MakeDeconstructionConstructionStep` constructs the final tuple type (returned by the deconstruction assignment) and checks constraints.

The tuple created by `MakeMergedTupleType` is not identical to that one (it represents a smaller tree of types, it is used for figuring out natural types in tuple literal). I think it is better not to check constraints here (it would report errors that are already reported later).

---
In reply to: [92523053](https://github.com/dotnet/roslyn/pull/15641#discussion_r92523053) [](ancestors = 92523053)